x-pack/plugins/security_solution/server/lib/detection_engine/signals/rule_status_service.test.ts

```diff
@@ -1,238 +0,0 @@
- * 2.0.
- */
-
-import {
-  buildRuleStatusAttributes,
-  RuleStatusService,
-  ruleStatusServiceFactory,
-  MAX_RULE_STATUSES,
-} from './rule_status_service';
-import { exampleRuleStatus, exampleFindRuleStatusResponse } from './__mocks__/es_results';
-import { RuleExecutionStatus } from '../../../../common/detection_engine/schemas/common/schemas';
-import { RuleExecutionLogClient } from '../rule_execution_log/__mocks__/rule_execution_log_client';
-import { UpdateExecutionLogArgs } from '../rule_execution_log/types';
-
-const expectIsoDateString = expect.stringMatching(/2.*Z$/);
-const buildStatuses = (n: number) =>
-  Array(n)
-    .fill(exampleRuleStatus())
-    .map((status, index) => ({
-      ...status,
-      id: `status-index-${index}`,
-    }));
-
-describe('buildRuleStatusAttributes', () => {
-  it('generates a new date on each call', async () => {
-    const { statusDate } = buildRuleStatusAttributes(RuleExecutionStatus['going to run']);
-    await new Promise((resolve) => setTimeout(resolve, 10)); // ensure time has passed
-    const { statusDate: statusDate2 } = buildRuleStatusAttributes(
-      RuleExecutionStatus['going to run']
-    );
-
-    expect(statusDate).toEqual(expectIsoDateString);
-    expect(statusDate2).toEqual(expectIsoDateString);
-    expect(statusDate).not.toEqual(statusDate2);
-  });
-
-  it('returns a status and statusDate if "going to run"', () => {
-    const result = buildRuleStatusAttributes(RuleExecutionStatus['going to run']);
-    expect(result).toEqual({
-      status: 'going to run',
-      statusDate: expectIsoDateString,
-    });
-  });
-
-  it('returns success fields if "success"', () => {
-    const result = buildRuleStatusAttributes(RuleExecutionStatus.succeeded, 'success message');
-    expect(result).toEqual({
-      status: 'succeeded',
-      statusDate: expectIsoDateString,
-      lastSuccessAt: expectIsoDateString,
-      lastSuccessMessage: 'success message',
-    });
-
-    expect(result.statusDate).toEqual(result.lastSuccessAt);
-  });
-
-  it('returns warning fields if "warning"', () => {
-    const result = buildRuleStatusAttributes(
-      RuleExecutionStatus.warning,
-      'some indices missing timestamp override field'
-    );
-    expect(result).toEqual({
-      status: 'warning',
-      statusDate: expectIsoDateString,
-      lastSuccessAt: expectIsoDateString,
-      lastSuccessMessage: 'some indices missing timestamp override field',
-    });
-
-    expect(result.statusDate).toEqual(result.lastSuccessAt);
-  });
-
-  it('returns failure fields if "failed"', () => {
-    const result = buildRuleStatusAttributes(RuleExecutionStatus.failed, 'failure message');
-    expect(result).toEqual({
-      status: 'failed',
-      statusDate: expectIsoDateString,
-      lastFailureAt: expectIsoDateString,
-      lastFailureMessage: 'failure message',
-    });
-
-    expect(result.statusDate).toEqual(result.lastFailureAt);
-  });
-});
-
-describe('ruleStatusService', () => {
```

Comment:
Unit tests for rule status service are deleted, but we don't have any for `SavedObjectsAdapter`.
Slightly concerning, but I'm ok with merging it.
We need to remember that this is our tech debt. We will need to cover the rule execution log implementation with tests after the migration to event_log.

Reply:
Yea, according to our plan, `SavedObjectsAdapter` will be removed altogether in 7.16. Otherwise, we'll need to cover adapters with tests.

Reply:
Sounds good to me 👍